.github/workflows/pr-notification-slack.yaml

```diff
@@ -0,0 +1,338 @@
+name: Fetch All Open and Recently Closed Pull Requests using cron job
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 10 * * 1-5'  # At 10:00 on every day-of-week from Monday through Friday(UTC Time), 7:30pm in IST.
+
+jobs:
+  fetch_all_pull_requests_and_notify_using_condition_and_cron:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Fetch recent cache
+        run: |
+          gh extension install actions/gh-actions-cache
+
+          echo "Fetching list of cache keys........"
+          cacheKeys=$(gh actions-cache list -R $REPO -B $BRANCH -L 100 | cut -f 1 )
+          echo "Cache keys-->: $cacheKeys"
+
+          # Print the contents of the cacheKeysForPR variable
+          echo "Contents of cacheKeys:"
+          echo "$cacheKeys"
+
+          # Extract the first two cache keys
+          mostRecentCacheKeys=$(echo "$cacheKeys" | head -n 1)
```

Comment:
Why are we taking recent two cache keys?

Reply:
The comment actually confused you, the code is correct, so I made the change in the comment.